test/parallel/test-dns-verbatim.js

```diff
@@ -0,0 +1,59 @@
+// Flags: --expose-internals
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { internalBinding } = require('internal/test/binding');
+const cares = internalBinding('cares_wrap');
+const { promisify } = require('util');
+
+// Test that changing require('dns').verbatim works as expected.
+
+const originalGetaddrinfo = cares.getaddrinfo;
+const calls = [];
+cares.getaddrinfo = common.mustCallAtLeast((...args) => {
+  calls.push(args);
+  originalGetaddrinfo(...args);
+}, 1);
+
+const dns = require('dns');
+const dnsPromises = dns.promises;
+
+let verbatim;
+
+// We want to test the parameter of verbatim only so that we
+// ignore possible errors here.
+function allowFailed(fn) {
+  return fn.catch((_err) => {
+    //
+  });
+}
+
+(async () => {
+  let callsLength = 0;
+  const checkParameter = (expected) => {
+    assert.strictEqual(calls.length, callsLength + 1);
+    verbatim = calls[callsLength][4];
+    assert.strictEqual(verbatim, expected);
+    callsLength += 1;
+  };
+
+  dns.verbatim = true;
```

Comment:
From a DX point of view, it's going to be a bit confusing for the command-line arg and the `dns.verbatim` API to use different value types (string labels vs. boolean). Also, the boolean here makes things a bit difficult if we decide to add a third option later. Perhaps instead of the boolean flag here it should use the same values as the command line arg? e.g. `dns.order = 'ipv4'` and `dns.order = 'verbatim'`, etc.

Reply:
Yes, they are different types(or they are not consistent enough).

> Perhaps instead of the boolean flag here it should use the same values as the command line arg? 

I guess I should have kept this as simple as a boolean value which only focuses on the parameter of  `verbatim`.

Reply:
Roll back to `--dns-verbatim` and I think it's much clear. 

As we can't disable `verbatim` with a boolean `--dns-verbatim`, it's still a string value .